data/base.go

```diff
@@ -0,0 +1,256 @@
+}
+
+func PayloadValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	//a place holder for more through validation
+	if _, ok := field.Interface().(interface{}); ok {
+		return true
+	}
+	return false
+}
+
+func AnnotationsValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	//a place holder for more through validation
+	if _, ok := field.Interface().([]interface{}); ok {
+		return true
+	}
+	return false
+}
+
+func TimezoneOffsetValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if offset, ok := field.Interface().(int); ok {
+		if offset >= tzValidationLowerLimit && offset <= tzValidationUpperLimit {
+			return true
+		}
+	}
+	return false
+}
+
+func TimeObjectValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if timeObject, ok := field.Interface().(time.Time); ok {
+		return isTimeObjectValid(timeObject)
+	}
+	return false
+}
+
+func isTimeObjectValid(timeObject time.Time) bool {
+	return !timeObject.IsZero() && timeObject.Before(time.Now())
+}
+
+func TimeStringValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if timeString, ok := field.Interface().(string); ok {
+		return isTimeStringValid(timeString)
+	}
+	return false
+}
+
+func isTimeStringValid(timeString string) bool {
+	var timeObject time.Time
+	timeObject, err := time.Parse(time.RFC3339, timeString)
+	if err != nil {
+		timeObject, err = time.Parse("2006-01-02T15:04:05", timeString)
+		if err != nil {
+			return false
+		}
+	}
+
+	return isTimeObjectValid(timeObject)
+}
+
+func ToString(fieldName string, data interface{}, errs validate.ErrorProcessing) *string {
+	if data == nil {
+		return nil
+	}
+	aString, ok := data.(string)
+	if !ok {
+		errs.Append(validate.NewPointerError(errs.BasePath+"/"+fieldName, "Invalid type", "should be of type string"))
+		return nil
+	}
+	return &aString
+}
+
+func ToFloat64(fieldName string, data interface{}, errs validate.ErrorProcessing) *float64 {
+	if data == nil {
+		return nil
+	}
+	theFloat, ok := data.(float64)
+	if !ok {
+		errs.Append(validate.NewPointerError(errs.BasePath+"/"+fieldName, "Invalid type", "should be of type float"))
+		return nil
+	}
+	return &theFloat
+}
+
+func ToInt(fieldName string, data interface{}, errs validate.ErrorProcessing) *int {
+	if data == nil {
+		return nil
+	}
+	theInt, ok := data.(int)
+
+	if !ok {
+		if theInt == 0 {
+			return &theInt
```

Comment:
Not sure I understand this logic.  I believe 0 is the value returned for `data.(int)` if ANY error occurs. It is the default value for an `int`.